pkg/detectors/azure_cosmosdb/azure_cosmosdb.go

```diff
@@ -0,0 +1,169 @@
+package azure_cosmosdb
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/cache/simple"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	dbKeyPattern = regexp.MustCompile(`([A-Za-z0-9+/]{86}==)`)
+	// account name can contain only lowercase letters, numbers and the `-` character, must be between 3 and 44 characters long.
+	accountUrlPattern = regexp.MustCompile(`([a-z0-9-]{3,44}.documents\.azure\.com)`)
+
+	invalidHosts = simple.NewCache[struct{}]()
+
+	noHostErr = errors.New("no such host")
+)
+
+func (s Scanner) getClient() *http.Client {
+	if s.client != nil {
+		return s.client
+	}
+
+	return defaultClient
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+func (s Scanner) Type() detectorspb.DetectorType {
+	return detectorspb.DetectorType_AzureCosmosDB
+}
+
+func (s Scanner) Description() string {
+	return "Azure Cosmos DB is a globally distributed, multi-model database service offered by Microsoft. CosmosDB keys and connection string are used to connect with Cosmos DB."
+}
+
+func (s Scanner) Keywords() []string {
+	return []string{".documents.azure.com"}
+}
+
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	var uniqueKeyMatches, uniqueAccountMatches = make(map[string]struct{}), make(map[string]struct{})
+
+	for _, match := range dbKeyPattern.FindAllStringSubmatch(dataStr, -1) {
+		uniqueKeyMatches[match[1]] = struct{}{}
+	}
+
+	for _, match := range accountUrlPattern.FindAllStringSubmatch(dataStr, -1) {
+		uniqueAccountMatches[match[1]] = struct{}{}
+	}
+
+	for key := range uniqueKeyMatches {
+		for accountUrl := range uniqueAccountMatches {
+			if invalidHosts.Exists(accountUrl) {
+				delete(uniqueAccountMatches, accountUrl)
+				continue
+			}
+
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_AzureCosmosDB,
+				Raw:          []byte(key),
```

Comment:
The URL should also be added. https://github.com/trufflesecurity/trufflehog/pull/3938#discussion_r1971982896